browser/DocumentRenderer.js

```diff
@@ -985,7 +986,12 @@ class DocumentRenderer extends DocumentRendererBase {
         watcherDefinition = watcherDefinition.apply(null, [attributes]);
       }
 
-      componentContext.watcher = this._state.getWatcher(watcherDefinition);
+      if (watcherDefinition) {
+        componentContext.watcher = this._state.getWatcher(watcherDefinition);
+      } else {
+        this._logger.warn(util.format(WARN_MISSING_WATCHER, watcherName));
```

Comment:
You get exception at render method, if you call `this.$context.getWatcherData()`.
You get same interface for component without watcher, and for component with missed watcher.